File: Cryptographic_Frontend/cf/SeedRandomMethod.cpp
#include "SeedRandomMethod.hpp"

#include "Argument.hpp"
#include "ResponseMessage.hpp"

#include <json/json.h>

using namespace cf;

SeedRandomMethod::SeedRandomMethod(std::string seed)
: Method("SeedRandom")
{
  addArgument(ArgumentPtr(new StringArgument("seed", seed)));
}

ResponseMessagePtr SeedRandomMethod::parseResponse(const std::string & message) {
  return ResponseMessagePtr(new ResponseMessage());
}

File: Cryptographic_Backend/src/main/java/cl/niclabs/cb/backend/methods/SeedRandomMethod.java
package cl.niclabs.cb.backend.methods;

import cl.niclabs.cb.backend.Method;

public interface SeedRandomMethod extends Method {
    static public class Args {
        public final String sessionHandler;
        public final String seed;

        public Args(String sessionHandler, String seed) {
            this.sessionHandler = sessionHandler;
            this.seed = seed;
        }
    }
}
